pyiron_base/storage/datacontainer.py

```diff
@@ -451,7 +430,7 @@ def to_builtin(self, stringify=False):
                 # _normalize calls int() on all digit string keys this is
                 # transparent for the rest of the module
                 k = str(k)
-                if isinstance(v, DataContainer):
+                if isinstance(v, type(self)):
```

Comment:
Is this (and the other nearby `type(self)` appearances) really correct? I naively would have thought this should be `isinstance(v, DataContainerBase))`, since that is already what guarantees us to have `.to_builtin`. 

What _should_ the behaviour be if a `DataContainer` instance has a `DataContainerBase` instance as an item? I would want it to "just work", but I know that is vague. Here I think it certainly wouldn't as the type check would evaluate `False` and we'd get kicked over to the `repr` branch.

I don't know what the exact right behaviour is, but I think this needs a test added to solidify whatever choice you make.

Reply:
Agree with your reasoning, it seems a more useful choice, I just hadn't considered it.  I've changed it back, but tests will have to come later.